c10/util/bit_cast.h

```diff
@@ -3,8 +3,18 @@
 #include <cstring>
 #include <type_traits>
 
+#if __has_include(<bit>) && (__cplusplus >= 202002L || (defined(__cpp_lib_bit_cast) && __cpp_lib_bit_cast >= 201806L))
+#include <bit>
+#define C10_HAVE_STD_BIT_CAST 1
+#else
+#define C10_HAVE_STD_BIT_CAST 0
```

Comment:
Similar logic to the prev PR. Does this macro serve any useful purpose. Main difference is whether bit_cast is constexpr I guess?

Reply:
Generally speaking, it would be good to move those `C10_HAVE_...` macros to say `c10/macros/Macros.h` or define new `Compiler.h` there

As defining those in various headers can lead to a potential duplication and than divergence

Reply:
I just wanted to avoid duplicating the condition in this file.

> defining those in various headers

what other header would reasonably define or consume this? this is the bit_cast header.

I can update the diff to undefine its temporary macro